### PR TITLE
Codegen debugging mapping

### PR DESCRIPTION
**Pull request by benibenj**

### CODE REVIEW

File: dace/sourcemap.py [hunk 1]
```diff
@@ -210,12 +210,18 @@ def create_cpp_map(code: str, name: str, target_name: str):
     # We send a message to the IDE in that case to notify the user of
     # restricted features.
     if Config.get('cache') != 'hash':
+        codegen_debug = True if Config.get('compiler',
+                                           'codegen_lineinfo') else False
```
Comment (benibenj) [inline dace/sourcemap.py:214]: Could Config.get('compiler','codegen_debug') be None? otherwise I can remove the if/else

File: dace/sourcemap.py [hunk 2]
```diff
@@ -237,15 +244,24 @@ def __init__(self, code: str, name: str, target_name: str):
...
-        """ For each line of code retrieve the corresponding nodes
-            and map the nodes to this line
+    def mapper(self, codegen_debug: bool = False):
+        """ For each line of code retrieve the corresponding identifiers
+            and create the mapping
         """
+        cpp_pattern = re.compile(
+            r'(\/\/\/\/__DACE:[0-9]+:[0-9]+:[0-9]+(,[0-9])*)')
+        codegen_pattern = re.compile(
+            r'(\/\/\/\/__CODEGEN;([A-z]:)?(\/|\\)([A-z0-9-_+]+(\/|\\))*([A-z0-9]+\.[A-z0-9]+);[0-9]+)'
```
Comment (benibenj) [inline dace/sourcemap.py:256]: I precompile the RE patterns now so it's only done once. I should have done this before

File: dace/sourcemap.py (hunk 1 above)
Comment (tbennun) [reply dace/sourcemap.py:214]: Please use `Config.get_bool`. This always ensures either true or false

File: dace/sourcemap.py (hunk 2 above)
Comment (tbennun) [reply dace/sourcemap.py:256]: nice :)

Comment (tbennun) [reply dace/sourcemap.py:256]: shouldn't this be part of `self`, though (as a field)? then you wouldn't need to pass it along to any method

Comment (benibenj) [reply dace/sourcemap.py:256]: Sure, could do this, does it make a difference? Just more readable code

Comment (tbennun) [reply dace/sourcemap.py:256]: yes, more readable code and fewer arguments (shorter diff too :) )

**Review by tbennun — APPROVED** [review]

**Pull request merged**